external/fv3fit/fv3fit/_shared/models.py

```diff
@@ -31,11 +31,16 @@ def __init__(
         """
         # if base_model is itself a DerivedModel, combine the underlying base model
         # and combine derived attributes instead of wrapping twice with DerivedModel
-        self.base_model: Predictor = model.base_model if isinstance(
-            model, DerivedModel
-        ) else model
+        if isinstance(model, DerivedModel):
+            self.base_model: Predictor = model.base_model
+            existing_derived_outputs = model._derived_output_variables  # type: ignore
```

Comment:
mypy complains that it doesn't know the type of the RHS, even though this is within an `isinstance` block.